File: samples/server/petstore/nodejs-google-cloud-functions/controllers/PetService.js
'use strict';

exports.addPet = function(args, res, next) {
  /**
   * Add a new pet to the store
   * 
   *
   * body Pet Pet object that needs to be added to the store
   * no response value expected for this operation
   **/
  res.end();
}

exports.deletePet = function(args, res, next) {
  /**
   * Deletes a pet
   * 
   *
   * petId Long Pet id to delete
   * api_key String  (optional)
   * no response value expected for this operation
   **/
  res.end();
}

exports.findPetsByStatus = function(args, res, next) {
  /**
   * Finds Pets by status
   * Multiple status values can be provided with comma separated strings
   *
   * status List Status values that need to be considered for filter
   * returns List
   **/
  var examples = {};
  examples['application/json'] = [ {
  "photoUrls" : [ "aeiou" ],
  "name" : "doggie",
  "id" : 0,
  "category" : {
    "name" : "aeiou",
    "id" : 6
  },
  "tags" : [ {
    "name" : "aeiou",
    "id" : 1
  } ],
  "status" : "available"
} ];
  if (Object.keys(examples).length > 0) {
    res.setHeader('Content-Type', 'application/json');
    res.end(JSON.stringify(examples[Object.keys(examples)[0]] || {}, null, 2));
  } else {
    res.end();
  }
}

exports.findPetsByTags = function(args, res, next) {
  /**
   * Finds Pets by tags
   * Multiple tags can be provided with comma separated strings. Use tag1, tag2, tag3 for testing.
   *
   * tags List Tags to filter by
   * returns List
   **/
  var examples = {};
  examples['application/json'] = [ {
  "photoUrls" : [ "aeiou" ],
  "name" : "doggie",
  "id" : 0,
  "category" : {
    "name" : "aeiou",
    "id" : 6
  },
  "tags" : [ {
    "name" : "aeiou",
    "id" : 1
  } ],
  "status" : "available"
} ];
  if (Object.keys(examples).length > 0) {
    res.setHeader('Content-Type', 'application/json');
    res.end(JSON.stringify(examples[Object.keys(examples)[0]] || {}, null, 2));
  } else {
    res.end();
  }
}

exports.getPetById = function(args, res, next) {
  /**
   * Find pet by ID
   * Returns a single pet
   *
   * petId Long ID of pet to return
   * returns Pet
   **/
  var examples = {};
  examples['application/json'] = {
  "photoUrls" : [ "aeiou" ],
  "name" : "doggie",
  "id" : 0,
  "category" : {
    "name" : "aeiou",
    "id" : 6
  },
  "tags" : [ {
    "name" : "aeiou",
    "id" : 1
  } ],
  "status" : "available"
};
  if (Object.keys(examples).length > 0) {
    res.setHeader('Content-Type', 'application/json');
    res.end(JSON.stringify(examples[Object.keys(examples)[0]] || {}, null, 2));
  } else {
    res.end();
  }
}

exports.updatePet = function(args, res, next) {
  /**
   * Update an existing pet
   * 
   *
   * body Pet Pet object that needs to be added to the store
   * no response value expected for this operation
   **/
  res.end();
}

exports.updatePetWithForm = function(args, res, next) {
  /**
   * Updates a pet in the store with form data
   * 
   *
   * petId Long ID of pet that needs to be updated
   * name String Updated name of the pet (optional)
   * status String Updated status of the pet (optional)
   * no response value expected for this operation
   **/
  res.end();
}

exports.uploadFile = function(args, res, next) {
  /**
   * uploads an image
   * 
   *
   * petId Long ID of pet to update
   * additionalMetadata String Additional data to pass to server (optional)
   * file File file to upload (optional)
   * returns ApiResponse
   **/
  var examples = {};
  examples['application/json'] = {
  "code" : 0,
  "type" : "aeiou",
  "message" : "aeiou"
};
  if (Object.keys(examples).length > 0) {
    res.setHeader('Content-Type', 'application/json');
    res.end(JSON.stringify(examples[Object.keys(examples)[0]] || {}, null, 2));
  } else {
    res.end();
  }
}

File: samples/server/petstore/java-helidon-server/se/src/main/java/org/openapitools/server/api/PetService.java
package org.openapitools.server.api;

import java.io.File;
import io.helidon.webserver.Handler;
import java.util.List;
import org.openapitools.server.model.ModelApiResponse;
import com.fasterxml.jackson.databind.ObjectMapper;
import org.openapitools.server.model.Pet;
import java.util.Set;

import io.helidon.webserver.Routing;
import io.helidon.webserver.ServerRequest;
import io.helidon.webserver.ServerResponse;
import io.helidon.webserver.Service;

public interface PetService extends Service { 

    /**
     * A service registers itself by updating the routing rules.
     * @param rules the routing rules.
     */
    @Override
    default void update(Routing.Rules rules) {
        rules.post("/pet", Handler.create(Pet.class, this::addPet));
        rules.delete("/pet/{petId}", this::deletePet);
        rules.get("/pet/findByStatus", this::findPetsByStatus);
        rules.get("/pet/findByTags", this::findPetsByTags);
        rules.get("/pet/{petId}", this::getPetById);
        rules.put("/pet", Handler.create(Pet.class, this::updatePet));
        rules.post("/pet/{petId}", this::updatePetWithForm);
        rules.post("/pet/{petId}/uploadImage", this::uploadFile);
        rules.post("/fake/{petId}/uploadImageWithRequiredFile", this::uploadFileWithRequiredFile);
    }


    /**
     * POST /pet : Add a new pet to the store.
     * @param request the server request
     * @param response the server response
     * @param pet Pet object that needs to be added to the store 
     */
    void addPet(ServerRequest request, ServerResponse response, Pet pet);

    /**
     * DELETE /pet/{petId} : Deletes a pet.
     * @param request the server request
     * @param response the server response
     */
    void deletePet(ServerRequest request, ServerResponse response);

    /**
     * GET /pet/findByStatus : Finds Pets by status.
     * @param request the server request
     * @param response the server response
     */
    void findPetsByStatus(ServerRequest request, ServerResponse response);

    /**
     * GET /pet/findByTags : Finds Pets by tags.
     * @param request the server request
     * @param response the server response
     */
    void findPetsByTags(ServerRequest request, ServerResponse response);

    /**
     * GET /pet/{petId} : Find pet by ID.
     * @param request the server request
     * @param response the server response
     */
    void getPetById(ServerRequest request, ServerResponse response);

    /**
     * PUT /pet : Update an existing pet.
     * @param request the server request
     * @param response the server response
     * @param pet Pet object that needs to be added to the store 
     */
    void updatePet(ServerRequest request, ServerResponse response, Pet pet);

    /**
     * POST /pet/{petId} : Updates a pet in the store with form data.
     * @param request the server request
     * @param response the server response
     */
    void updatePetWithForm(ServerRequest request, ServerResponse response);

    /**
     * POST /pet/{petId}/uploadImage : uploads an image.
     * @param request the server request
     * @param response the server response
     */
    void uploadFile(ServerRequest request, ServerResponse response);

    /**
     * POST /fake/{petId}/uploadImageWithRequiredFile : uploads an image (required).
     * @param request the server request
     * @param response the server response
     */
    void uploadFileWithRequiredFile(ServerRequest request, ServerResponse response);

}
